### PR TITLE
feat: add either interface on restclients with all response status codes

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/5" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/5?utm_source=stack-comment).

File: src/lib/primitives/jsonschema/jsonschema.ts
Comment (MickVanDuijn) [inline src/lib/primitives/jsonschema/jsonschema.ts:235]: decided not to use `axioms` in the end?

File: test/petstore.spec.ts
Comment (MickVanDuijn) [inline test/petstore.spec.ts:20]: how come `right` is not triggering an error here? :thinking_face: there is no `isRight` guard around it

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/5?utm_source=stack-comment).


Job Reference: f0penGwW85vorg9X6nFC

**Pull request merged**
